crates/sui-transactional-test-runner/src/test_adapter.rs

```diff
@@ -624,11 +632,163 @@ impl<'a> MoveTestAdapter<'a> for SuiTestAdapter<'a> {
                 let output = self.object_summary_output(&summary, view_events, view_gas_used);
                 Ok(output)
             }
+            SuiSubcommand::UpgradePackage(UpgradePackageCommand {
+                package,
+                upgrade_capability,
+                dependencies,
+                sender,
+                gas_budget,
+                syntax,
+            }) => {
+                let syntax = syntax.unwrap_or_else(|| self.default_syntax());
+                let data = data.ok_or_else(|| {
+                    anyhow::anyhow!(
+                        "Expected a module text block following 'upgrade' starting on lines {}-{}",
+                        start_line,
+                        command_lines_stop
+                    )
+                })?;
+
+                let state = self.compiled_state();
+                let package_id = state.resolve_named_address(package.as_str()).into();
+                // Reset package address to 0x0 which resets module addressses to 0x0.
+                *state.named_address_mapping.get_mut(&package).unwrap() =
```

Comment:
What happens if the upgrade fails?  In the "real world" you would be able to continue using the old package, but with this change, in the transactional test setting, the named address entry gets wiped so that future transactions that wanted to use it would fail.  This seems bad to me.

Also, we want to support tests where both the old and new versions of a package are accessed in a transaction.  I'm assuming from this setup that we expect the upgrade to overwrite the named address entry for the package it is upgrading, which (if I understand correctly) makes that impossible.

I had previously suggested a solution where only the local version of the named address mapping gets updated, and the upgrade transaction accepts two names -- a mandatory name for the package being upgraded and an optional name to give the package being upgraded to -- did that not work?